File: lab-go/internal/provider/codex.go
package provider

import (
	"context"
	"fmt"
)

// Codex is a stub for the OpenAI Codex CLI.
type Codex struct {
	base
}

func (c *Codex) Name() string { return "codex" }

func (c *Codex) Run(_ context.Context, query string) (RunResult, error) {
	return RunResult{}, fmt.Errorf("%w: Codex provider is a stub. Implement Run() against the Codex CLI, then detect skill %q tool-use. Query was: %q",
		ErrNotImplemented, c.cfg.SkillName, query)
}

File: lab/providers/codex.py
"""Codex CLI provider stub."""

from __future__ import annotations

from .base import Provider, ProviderNotImplementedError, RunResult


class CodexProvider(Provider):
    """
    Contract for OpenAI Codex CLI.

    Expected invocation (to implement):
      codex exec --model <model> <query>
      (or equivalent non-interactive Codex command)

    Trigger detection should look for skill tool-use markers for `skill_name`
    in stdout/stderr (JSON events or Skill "name" style logs).
    """

    name = "codex"

    def run(self, query: str) -> RunResult:
        raise ProviderNotImplementedError(
            "Codex provider is a stub. Implement run() against the Codex CLI, "
            f"then detect skill {self.skill_name!r} tool-use in the output. "
            f"Query was: {query!r}"
        )
